src/main/java/com/github/shiranr/scooters/db/Client.java

```diff
@@ -0,0 +1,89 @@
+package com.github.shiranr.scooters.db;
+
+import com.azure.cosmos.ConsistencyLevel;
+import com.azure.cosmos.CosmosClient;
+import com.azure.cosmos.CosmosClientBuilder;
+import com.azure.cosmos.CosmosContainer;
+import com.azure.cosmos.CosmosDatabase;
+import com.azure.cosmos.models.CosmosItemResponse;
+import com.azure.cosmos.models.CosmosQueryRequestOptions;
+import com.azure.cosmos.util.CosmosPagedIterable;
+import com.github.shiranr.scooters.domain.Scooter;
+
+import java.util.Optional;
+
+import static java.net.HttpURLConnection.HTTP_ACCEPTED;
+import static java.net.HttpURLConnection.HTTP_CREATED;
+import static java.net.HttpURLConnection.HTTP_OK;
+
+/**
+ * This class is singleton. The goal is to have a managed client for our service.
+ * Currently, this is not thread safe. Will be done later if needed.
+ * The DB HOST and PASSWORD are currently in ENV variables, should be moved to key vault.
+ */
+public class Client {
+    private static CosmosClient cosmos;
+    private static Client client;
+
+    static String DB_NAME = "lime";
+    static String CONTAINER_NAME = "scooters";
+
+    private Client() {
+        if (cosmos == null) {
+            String endpoint = System.getenv("DB_HOST");
+            String password = System.getenv("DB_PASSWORD");
+            cosmos = new CosmosClientBuilder()
+                    .endpoint(endpoint)
+                    .key(password)
+                    .consistencyLevel(ConsistencyLevel.EVENTUAL)
+                    .buildClient();
+        }
+    }
+
+    public static Client GetClient() {
+        if (client == null) {
+            client = new Client();
+        }
+        return client;
+    }
+
+    public Scooter[] GetAllScooters() {
```

Comment:
It looks like this is a concrete implementation, where the function contains "selects".
I believe it's more proper to convert to interface implementation,
otherwise this is not a testable code.
Same for the rest of the code.
This class should implement an interface and be injected where needed.

Reply:
It can also be done in the next iteration, as long as the technical debt is clearly stated in a TODO